core/templates/dev/head/pages/exploration_editor/simple_editor_tab/ScrollSyncDirective.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive used to mark parts of the editor that have a sidebar
+ * item associated to it, see ScrollSyncService.js.
+ */
+
+oppia.directive('scrollSync', ['ScrollSyncService', 'QuestionIdService',
+  function(ScrollSyncService, QuestionIdService) {
+  return {
+    restrict: 'A',
+    link: function(scope, element, attrs) {
+      if (!(element.prop('tagName') === 'H3')) {
+        console.log(element.tag, attrs.scrollSync);
+        throw Error('ScrollSync is only used with h3 elements.');
+      };
+      // Set the element Id.
+      var subfieldLabels = [
+        'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
+      var label = attrs.scrollSync;
+      var id;
+      if (!(subfieldLabels.indexOf(label) === -1)) {
+        var questionId = scope.questionId || scope.question.getId();
```

Comment:
This issue arise from trying to make the directive less verbose in usage, because i noticed that the questionId or question object is already available in the scope of the element on which scroll-sync is applied and instead of passing the questionId as an attribute i instead get it from the element scope.

scope.questionId is for cases like multiple_choice_editor_directive.html and scope.question is for simple_editor_body_directive.html.

But i can explicitly ask for the questionId doing something like this:

    <anchor-header text="Multiple choice" question-id="<[questionId]>" label="Multiple choice">

Or like this:

    <anchor-header text="Multiple choice" id="<getSubfieldId('Multiple Choice')">

where 'getSubfieldId' is a method defined on the scope, which will have to be repeated in future questions types directives.